cmd/kubeadm/app/util/runtime.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	kubeadmapiv1alpha2 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha2"
+	utilsexec "k8s.io/utils/exec"
+)
+
+// ContainerRuntime interface
+type ContainerRuntime interface {
+	IsDocker() bool
+	IsRunning() error
+	ListContainers() ([]string, error)
```

Comment:
I'd rename this `ListKubeContainers()` so it's clear that this is specifically about the Kubernetes-managed containers